swarm/storage/resource.go

```diff
@@ -91,186 +120,170 @@ type resource struct {
 // stored using a separate store, and forwarding/syncing protocols carry per-chunk
 // flags to tell whether the chunk can be validated or not; if not it is to be
 // treated as a resource update chunk.
-
-type ResourceValidator interface {
-	isOwner(string) (bool, error)
-	nameHash(string) common.Hash
-}
-
+//
+// TODO: Include modtime in chunk data + signature
 type ResourceHandler struct {
 	ChunkStore
+	ctx          context.Context
```

Comment:
We should comment why we are using `context.Context` in a non-typical way (and warn users to change their expectations).